### PR TITLE
[FLINK-18764][python] Add DataStream class and support from_collectio…

### PR DESCRIPTION
**Pull request by shuiqiangchen**
## What is the purpose of the change

Introduced a new class named DataStream including its common interfaces. And support to get a DataStream through StreamExecutionEnvironment.from_collection(). 

## Brief change log

- add a new python module named DataStream.
- support to get a DataStream through StreamExecutionEnvironment.from_collection()

## Verifying this change

This change is verified by the test cases in test_stream_execution_environment.py

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): ( no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn/Mesos, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (yes)
  - If yes, how is the feature documented? ( not documented)

### CODE REVIEW

File: flink-python/pyflink/datastream/tests/test_stream_execution_environment.py [hunk 1]
```diff
@@ -22,9 +22,12 @@
 
 import unittest
 
+
```
Comment (hequn8128) [inline flink-python/pyflink/datastream/tests/test_stream_execution_environment.py:25]: Please be careful about these unnecessary changes.

File: flink-python/pyflink/datastream/tests/test_stream_execution_environment.py [hunk 2]
```diff
@@ -211,3 +214,31 @@ def test_execute(self):
         self.assertEqual(len(execution_result.get_all_accumulator_results()), 0)
         self.assertIsNone(execution_result.get_accumulator_result('accumulator'))
         self.assertIsNotNone(str(execution_result))
+
+    def test_from_collection_without_data_types(self):
+        ds = self.env.from_collection([(1, 'Hi', 'Hello'), (2, 'Hello', 'Hi')])
+        test_sink = DataStreamTestCollectSink(True)
+        ds._j_data_stream.addSink(test_sink._j_data_stream_test_collect_sink)
```
Comment (hequn8128) [inline flink-python/pyflink/datastream/tests/test_stream_execution_environment.py:221]: There are some problems with the design:
1. we need to pass java datastream object explicitly which should be avoided.
2. we need to specify True false manually. I think we can infer the true false from the input datastream, i.e., set true when output_type is `PickledBytesTypeInfo`.

The code may looks like:
```
collect_util = DataStreamCollectUtil()
collect_util.collect(ds)
results = collect_util.results()
```

File: flink-python/pyflink/datastream/tests/test_util.py [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+#  See the License for the specific language governing permissions and
+# limitations under the License.
+################################################################################
+
+import pickle
+
+from pyflink.java_gateway import get_gateway
+
+
+class DataStreamTestCollectSink(object):
```
Comment (hequn8128) [inline flink-python/pyflink/datastream/tests/test_util.py:24]: Refactor the class according to the comments above.

File: flink-python/pyflink/datastream/data_stream.py [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+    def get_name(self) -> str:
+        """
+        Gets the name of the current data stream. This name is used by the visualization and logging
+        during runtime.
+
+        :return: Name of the stream.
+        """
+        return self._j_data_stream.getName()
+
+    def name(self, name: str):
```
Comment (hequn8128) [inline flink-python/pyflink/datastream/data_stream.py:43]: Should also add tests to verify these methods.

Comment (shuiqiangchen) [reply flink-python/pyflink/datastream/data_stream.py:43]: Yes, I will add the corresponding tests to verify them.

File: flink-python/pyflink/datastream/tests/test_stream_execution_environment.py (hunk 1 above)
Comment (shuiqiangchen) [reply flink-python/pyflink/datastream/tests/test_stream_execution_environment.py:25]: Thank you, I will pay more attention to it.

File: flink-python/pyflink/datastream/tests/test_stream_execution_environment.py (hunk 2 above)
Comment (shuiqiangchen) [reply flink-python/pyflink/datastream/tests/test_stream_execution_environment.py:221]: Yes, there might be some spaces for improvements.

File: flink-python/pyflink/datastream/tests/test_util.py (hunk 3 above)
Comment (shuiqiangchen) [reply flink-python/pyflink/datastream/tests/test_util.py:24]: Ok, I will do it.

**Comment by shuiqiangchen** [comment]
Hi @hequn8128 , thank you for your comments, I have updated the pr according to your suggestions, please have a look at it.

**Comment by hequn8128** [comment]
@shuiqiangchen Thanks a lot for the update. LGTM, will merge this pr after test passed.
